### ISSUE TITLE
Problem with Makefile.PL && mysql_config on Windows

### ISSUE THREAD

**Issue by mbeijen** [perl5-dbi/DBD-mysql#96]
After @pali stored `mysql_config`s value in 41263aa1afe3e121ca762ce9667c4dc325e39eb9 I have this issue:

StrawberryPerl provides its own batch script that does more or less what `mysql_config` does:

    C:\Users\michi\DBD-mysql>where mysql_config
    C:\berrybrew\5.24.0_64\c\bin\mysql_config.bat

The first time I run `perl Makefile.PL` I get a message that `mysql_config` can't be found, but it *does* work:
```
C:\Users\michi\DBD-mysql>perl Makefile.PL

Cannot find the file 'mysql_config'! Your execution PATH doesn't seem
to contain the path to mysql_config. Resorting to guessed values!
<snip>
I will use the following settings for compiling and testing:

  cflags        (mysql_config) = -IC:\berrybrew\5.24.0_64\c\bin\..\include\mysql616
  embedded      (guessed     ) =
  ldflags       (guessed     ) =
  libs          (mysql_config) = -LC:\berrybrew\5.24.0_64\c\bin\..\lib -lmysql
  mysql_config  (guessed     ) = mysql_config
<snip>
Checking if libs are available for compiling...
Looks good.
```

However, the *second* time around I run `perl Makefile.PL`,  `t\mysql.mtest` has a value for `mysql_config` and everything fails:

```
C:\Users\michi\DBD-mysql>perl Makefile.PL
Use of uninitialized value in concatenation (.) or string at Makefile.PL line 79.
'--version' is not recognized as an internal or external command,
operable program or batch file.
<snip>
Use of uninitialized value in concatenation (.) or string at Makefile.PL line 576.
'--cflags' is not recognized as an internal or external command,
operable program or batch file.
```

Also, `dmake clean` does not clean `t\mysql.mtest` and `dmake realclean` or `dmake clean` does not work after I ran `Makefile.PL` for the second time.

**Comment by pali** [comment]
Can you try to change 33th line in Makefile.PL from `$opt->{'mysql_config'} = $::test_mysql_config;` to `$opt->{'mysql_config'} = $::test_mysql_config if defined $::test_mysql_config;`? Or maybe defined and length?

**Comment by mbeijen** [comment]
That fixes only a part of the problem.
I think we should only store values in `mysql.mtest` if they're user provided. Also, the test for if `mysql_config` exists is broken on Windows, because we don't test if `Win32::GetShortPathName()` returns anything.

**Comment by pali** [comment]
Hm, right it make sense to store mysql_config value only if it was provided by user.

**Comment by mbeijen** [comment]
And also, I think Makefile.PL should exit if you supply a mysql_config that
does not exist.


Op zo 5 feb. 2017 om 14:46 schreef pali <notifications@github.com>

**Comment by pali** [comment]
Yes, make sense.

**Comment by pali** [comment]
Please test pull request #100 if it fixes all these problems...

**Comment by pali** [comment]
Fixed in 4.042.

**Comment by mbeijen** [comment]
Thanks!
